cfme/infrastructure/host.py

```diff
@@ -452,7 +448,12 @@ class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
+        from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+        dest = 'Hosts'
+        if hasattr(self.obj, 'provider') and self.obj.provider:
```

Comment:
then try/except may be a better and more pythonic choice?

We should move away from class navigation in future anyway having the collections in place. But that is for future :)